versioning/BRM/extentmap.cpp

```diff
@@ -1802,26 +1802,28 @@ void ExtentMap::save(const string& filename)
 
 #endif
 
-  grabEMEntryTable(READ);
-  grabEMIndex(READ);
+  // MCOL-5623 Grag the `WRITE` lock, we cannot let BRM sctructres to be modified (new Extent), while saving
+  // them to the file.
+  grabEMEntryTable(WRITE);
```

Comment:
READ lock taken assumes there will be no writes. 
Taking a WRITE lock you disallows other threads to read EM or EMIndex.

Reply:
I see, thanks for showing this, fixed